### PR TITLE
Reorganize azure testing

### LINKED ISSUE #1060: Build docs in Azure
## Tell us about it

Move automatic generation of documentation to Azure. Related to #896, as explained there, the solution should ideally build development version docs and (given some condition, github tag...) build the release docs.

### LINKED ISSUE #1061: Add coverage check in PRs
## Tell us about it
Add coverage check for PRs via Azure (either Azure alone if it were to work, via coveralls or codecov).

### PR DESCRIPTION
**Pull request by ahartikainen**
## Description

Implements all the tests and builds on Azure. Removes Travis. Fixes #1061 and fixes #1060

#### Python 3.6, 3.7, 3.8
- BaseTests --> Tests ArviZ based tests (stats, diagnostics, plots)
- ExternalTests --> Test external libraries (data)
- Run benchmarks --> Upload results to pipeline site

#### Python 3.8
- Build Docs (build only if tests are passed and on master branch)
- Build Wheel (build only if tests are passed and a tag "branch") --> upload `dist` folder to pipeline site --> can be download and upload with `twine` for pypi manually.

Changes coveralls to codecov.

## Checklist

- [x] Changes are listed in [changelog](https://github.com/arviz-devs/arviz/blob/master/CHANGELOG.md#v0xx-unreleased)

## Future

We could follow benchmarks somehow, but currently they are only there so we can follow them manually.

### CODE REVIEW

File: mytests.txt [hunk 1]
```diff
@@ -0,0 +1,42 @@
+============================= test session starts =============================
```
Comment (OriolAbril) [inline mytests.txt]: Is this file supposed to be here?

Comment (ahartikainen) [reply mytests.txt]: no, just my debugging

File: requirements-docs.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+ghp-import
+ipython
+nbsphinx
+numpydoc
+pydocstyle<5.0
```
[the remark this answers is not in the document]
Comment (ahartikainen) [reply requirements-docs.txt]: cool

**Comment by OriolAbril** [comment]
This is probably a good time to remove the `azurepipelines-coverage` file which doesn't actually do anything.

**Comment by ahartikainen** [comment]
Cool,

Still needs the docs

    .azure-pipelines/azure-pipelines-docs.yml

Some fix for coverage and pytest reports.

Also reporting lint etc?

Codecov? We probably should also skip tests folder from coverage.

I dropped some tests (saving external file)

**Comment by canyon289** [comment]
I am very late to this PR but just wanted to say the work here looks great! Was looking at the testing now and everything is much better

**Pull request merged**